libparsec/crates/client/src/certif/device.rs

```diff
@@ -0,0 +1,175 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_client_connection::{AuthenticatedCmds, ConnectionError};
+use libparsec_platform_storage::certificates::GetCertificateError;
+use libparsec_protocol::authenticated_cmds::v4::device_create;
+use libparsec_types::{
+    anyhow, thiserror, Bytes, CertificateSignerOwned, DateTime, DeviceCertificate, LocalDevice,
+    MaybeRedacted, SigningKeyAlgorithm, UserID,
+};
+
+use super::{
+    encrypt::CertifEncryptForUserError, greater_timestamp, CertificateBasedActionOutcome,
+    GreaterTimestampOffset, InvalidCertificateError,
+};
+
+pub async fn register_new_device(
```

Comment:
Given this code is only used by `Client::export_recovery_device` you may want to put it in the client/recovery_device.rs` module I was talking about in another comment.

I guess we could argue this code is fine here (given we are producing a certificate, so it's kind of okay to do that in the `certif` module), however currently we generate the user&device certificates directly in the code that need it (i.e. bootstrap organization and user/device greeter)  so we would stay consistent.

The `certif` module has currently a single goal: implement `CertificateOps`, however `registered_new_device` is not related to `CertificateOps` (given we don't need knowledge on the previous certificates to upload a new user/device certificate, which is not the case for e.g. `revoke_user` that hence is part of `CertificateOps`)

On top of that `register_new_device` is kind of broad (in a similar way to `LocalDevice::generate_new_device`), so I think it's better to keep in private in a dedicated module that only has a single goal, instead of having it public and risking to have it re-used later on for something it is not designed for...